tests/shared/nlu/training_data/test_training_data.py

```diff
@@ -637,6 +641,36 @@ def test_fingerprint_is_same_when_loading_data_again():
     assert td1.fingerprint() == td2.fingerprint()
 
 
+def test_persist_load_training_data_chunk(tmp_path: Path):
+    messages = [
+        Message(
+            features=[
+                Features(
+                    np.random.random([4, 3]), FEATURE_TYPE_SEQUENCE, TEXT, "spacy"
+                ),
+                Features(
+                    scipy.sparse.coo_matrix(np.random.randint(0, 10, [4, 3])),
+                    FEATURE_TYPE_SEQUENCE,
+                    TEXT,
+                    "regex",
+                ),
+            ]
+        )
+    ]
+
+    training_data_chunk = TrainingDataChunk(messages)
+    original_fingerprint = training_data_chunk.fingerprint()
+    file_path = training_data_chunk.persist_chunk(str(tmp_path), "test.tfrecord")
+
+    loaded_training_data_chunk = TrainingDataChunk.load_chunk(file_path)
```

Comment:
Optional: Should we test for the expected keys as well in the persisted tf record file?

Reply:
any suggestion on how we should do that? we cannot just read the content of the file into a string and check if the expected keys exists. also this is kind of already checked as loading would fail if the keys are not present.

Reply:
> this is kind of already checked as loading would fail if the keys are not present.

Makes sense, but just to be sure maybe add tests for just `_deconstruct_tf_record_key` and `_construct_tf_record_key` functions since they are static functions?